File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        stack<char> helperStack;
        for(char i : s){
            if((i == 'B' && !helperStack.empty() && helperStack.top() == 'A') 
                || (i == 'D' && !helperStack.empty() && helperStack.top() == 'C')){
                helperStack.pop();
            }else{
                helperStack.push(i);
            }
        }

        return helperStack.size();
    }
};

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.rs
impl Solution {
    pub fn min_length(s: String) -> i32 {
        let mut stack = Vec::new();
        
        for ch in s.chars() {
            if !stack.is_empty() && ((ch == 'B' && *stack.last().unwrap() == 'A') || (ch == 'D' && *stack.last().unwrap() == 'C')) {
                stack.pop();
            } else {
                stack.push(ch);
            }
        }
        
        stack.len() as i32
    }
}